tools/contractdeployer/contract_deployer.go

```diff
@@ -203,25 +206,22 @@ func signAndSendTxWithReceipt(wallet wallet.Wallet, client ethadapter.EthClient,
 	}
 
 	var start time.Time
-	var receipt *types.Receipt
-	var contractAddr *common.Address
 	for start = time.Now(); time.Since(start) < timeoutWait; time.Sleep(retryInterval) {
-		receipt, err = client.TransactionReceipt(signedTx.Hash())
+		receipt, err := client.TransactionReceipt(signedTx.Hash())
+		if err != nil {
+			log.Info(err.Error())
+		}
 		if err == nil && receipt != nil {
 			if receipt.Status != types.ReceiptStatusSuccessful {
 				return nil, fmt.Errorf("unable to deploy contract, receipt status unsuccessful: %v", receipt)
 			}
 			log.Info("Contract successfully deployed to %s", receipt.ContractAddress)
-			contractAddr = &receipt.ContractAddress
-			break
+			return &receipt.ContractAddress, nil
 		}
 
 		log.Info("Contract deploy tx has not been mined into a block after %s...", time.Since(start))
 	}
-	if contractAddr == nil {
-		return nil, fmt.Errorf("failed to mine contract deploy tx %s into a block after %s. Aborting", signedTx.Hash(), time.Since(start))
-	}
-	return contractAddr, nil
+	return nil, fmt.Errorf("failed to mine contract deploy tx %s into a block after %s. Aborting", signedTx.Hash(), time.Since(start))
```

Comment:
I think the idiomatic Go style tries to keep the happy path as the final return but this is a weird one and I don't feel strongly.

Reply:
Cool. I'll leave it for now